### PR TITLE
Bookstore: add reducers and actions

### PR DESCRIPTION
**Pull request by raminka13**
## Redux Toolkit

- Added Redux Toolkit.
- Structured folder tree to follow [ducks pattern](https://github.com/erikras/ducks-modular-redux) for Redux files.
- Wrote book actions and reducer.
- Wrote categories actions and reducer.
- Configured the Redux Store.

### CODE REVIEW

File: src/redux/books/book.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+const ADD = 'ADD';
+const REMOVE = 'REMOVE';
```
Comment (sumairq) [inline src/redux/books/book.js]: - [x] The value of each action type needs to follow the [ducks pattern](https://github.com/erikras/ducks-modular-redux), So please edit all action types to be something like this:
`const ADD_BOOK = bookstore/books/ADD_BOOK`

File: src/redux/categories/categories.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+const CHECK_STATUS = 'CHECK_STATUS';
```
Comment (sumairq) [inline src/redux/categories/categories.js]: - [x] Same here, the value of each action type needs to follow the [ducks pattern](https://github.com/erikras/ducks-modular-redux), So please edit all action types to be something like this:
`const ADD_BOOK = bookstore/books/ADD_BOOK`

**Review by sumairq — CHANGES_REQUESTED** [review]
Hi @raminka13 ,

 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ Good commit history
✔️Linters are passing
✔️Very professional README
✔️Correct Git flow
✔️Descriptive PR


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/book.js (hunk 1 above)
Comment (raminka13) [reply src/redux/books/book.js]: @sumairq **Thanks for great feedback**

File: src/redux/categories/categories.js (hunk 2 above)
Comment (raminka13) [reply src/redux/categories/categories.js]: @sumairq **Very kind of you to explain the missing pattern. Many thanks**

**Review by GzimAsani — APPROVED** [review]
# STATUS: APPROVED :100:         
## There is nothing else to say than it's time to merge... :partying_face: 
Cheers and Happy coding! :clap:  :computer:  :wine_glass: 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


## Optional suggestions
-N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a 

### Please be reminded to not open a new pull request after you get a review, only use the same one :thought_balloon: 


 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by raminka13** [comment]
Thanks, guys.
@sumairq 
@GzimAsani

**Pull request merged**